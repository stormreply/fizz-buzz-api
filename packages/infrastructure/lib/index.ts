import { Code, Function, Runtime } from '@aws-cdk/aws-lambda';
import { Construct, Stack, StackProps } from '@aws-cdk/core';

import path from 'path';
export class FizzBuzzApiStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    new Function(this, 'FizzBuzzAPILambda', {
      runtime: Runtime.NODEJS_14_X,
      code: Code.fromAsset(path.dirname(require.resolve('@stormreply/fizz-buzz-api-lambda/build/package.json'))),
      handler: 'src/index.default',
    });
  }
}
