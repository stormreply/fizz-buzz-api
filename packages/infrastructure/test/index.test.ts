import { App } from '@aws-cdk/core';

import 'jest-cdk-snapshot';

import { FizzBuzzApiStack } from '../lib';

test('Fizz Buzz API Stack', () => {
  expect(new FizzBuzzApiStack(new App(), 'FizzBuzzTestStack')).toMatchCdkSnapshot({
    ignoreAssets: true,
  });
});
