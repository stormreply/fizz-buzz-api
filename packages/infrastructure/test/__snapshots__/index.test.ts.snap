// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fizz Buzz API Stack 1`] = `
Object {
  "Parameters": Any<Object>,
  "Resources": Object {
    "FizzBuzzAPILambda7753EAB2": Object {
      "DependsOn": Array [
        "FizzBuzzAPILambdaServiceRole0A3C0CCB",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Handler": "src/index.default",
        "Role": Object {
          "Fn::GetAtt": Array [
            "FizzBuzzAPILambdaServiceRole0A3C0CCB",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "FizzBuzzAPILambdaServiceRole0A3C0CCB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
